### PR TITLE
Update sample app to support 'agent' scope

### PR DESCRIPTION
**Pull request by maria-robobug**
- Updates the travel-sample app to use the new `agent` scopes in Couchbase Server 7.0.

- Uses docker-compose to spin up Couchbase Server alongside the backend and FE apps

- Refactors some endpoints where necessary to accommodate new multi-tenant data model

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,33 +1,17 @@
-# Set the base image to Ubuntu
-FROM ubuntu
+FROM python:3.9-slim-buster
 
-# File Author / Maintainer
-MAINTAINER Couchbase
```
Comment (osfameron) [inline Dockerfile:5]: should we retain the MAINTAINER here? I don't know what this is used for.

Comment (maria-robobug) [reply Dockerfile:5]: `MAINTAINER` seems deprecated 🤷‍♀️ (my Vscode extension pointed out) , https://docs.docker.com/engine/reference/builder/#maintainer-deprecated

I think I can replace with LABEL though 👍

File: travel.py [hunk 2]
```diff
@@ -36,7 +38,9 @@
 if args.cluster:
     CONNSTR = "couchbase://" + args.cluster
 else:
-    CONNSTR = "couchbase://localhost"
+    # 'db' is an alias that resolves to the couchbase-server docker hostname.
+    # See `docker-compose.yml (line 25)`.
+    CONNSTR = "couchbase://db"
```
Comment (osfameron) [inline travel.py:40]: Nitpick: here and below for username/password, the if block seems unnecessary as you could set default in ArgParser instead. (But I realise this was in original, and maybe explicit is more readable...)

File: travel.py [hunk 3]
```diff
@@ -45,33 +49,37 @@
...
-def index():
-    return redirect("/index.html", code=302)
+CORS(app, headers=['Content-Type', 'Authorization'])
 
 
-@app.route('/<path:path>')
-def serveFiles(path):
-    return send_from_directory("static", path)
-
+@app.route('/')
```
Comment (osfameron) [inline travel.py:58]: Might be worth a docstring for this route.

File: travel.py [hunk 4]
```diff
@@ -141,16 +146,21 @@ def findall(self, fromloc, toloc):
...
+class TenantUserView(FlaskView):
+    """Class for storing user related information for a given tenant"""
 
-    @route('/login', methods=['POST', 'OPTIONS'])
-    def login(self):
+    @api.route('/tenants/<tenant>/user/login', methods=['POST', 'OPTIONS'])
+    @cross_origin(supports_credentials=True)
+    def login(tenant):
         """Login an existing user"""
+        agent = tenant.lower()
```
Comment (osfameron) [inline travel.py]: worth using the `make_user_key` function instead of calling lower directly?

File: travel.py [hunk 5]
```diff
@@ -45,33 +49,37 @@
...
 
-@app.route('/<path:path>')
-def serveFiles(path):
-    return send_from_directory("static", path)
-
+@app.route('/')
+def index():
+    endpoints = []
+    for rule in app.url_map.iter_rules():
+        endpoints.append(str(rule))
```
Comment (osfameron) [inline travel.py]: might be clearer as list comprehension (I was restraining myself, but I saw you use one later on line 296 so...)

**Comment by osfameron** [comment]
This looks great 👍 ! Now I have to go read up on Blueprints in Flask...

File: travel.py (hunk 5 above)
Comment (maria-robobug) [reply travel.py]: My Go brain should be switched off when I do Python :D

File: travel.py (hunk 2 above)
Comment (maria-robobug) [reply travel.py:40]: I think I will leave this as explicit, prefer it to be more readable

File: travel.py [hunk 6]
```diff
@@ -221,7 +238,11 @@ def flights(tenant, username):
                 for key in booked_flights:
                     rows.append(flights.get(key).content_as[dict])
                 print(rows)
-                respjson = jsonify({"data": rows})
+
+                querytype = "KV get - scoped to " + scope.name + \
```
Comment (osfameron) [inline travel.py]: I love the new output from the backend!

Python's string.format might be more idiomatic/readable here btw? Forget details, but something like:
`"scoped to {name}.user: for {num} bookings in document".format(name=..., num=...)`

Comment (maria-robobug) [reply travel.py]: Roger roger! 🤖

**Comment by osfameron** [comment]
LGTM 👍 with only one minor/subjective "polish" comment

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      - CB_HOST
+      - CB_USER
+      - CB_PSWD
+    command: $CB_HOST $CB_USER $CB_PSWD
+    volumes:
+      - .:/app
+    container_name: try-cb-api
+  
+  frontend:
+      build: "https://github.com/couchbaselabs/try-cb-frontend-v2#master"
```
Comment (osfameron) [inline docker-compose.yml:21]: I think this will become the 7.0 branch rather than master?
Just doing some tidying up of frontend and will merge soon btw.

**Pull request merged**